NOX: WIP add Tpetra implementation

maartenarnst opened the pull request:
@trilinos/nox
@rppawlo 

This PR is a WIP to add a `Tpetra` implementation of the `NOX` abstract classes `Group`,  `Vector`, `MultiVector`, ...

Proposed steps:
- [x] basic ETI infrastructure (the draft follows the design of `Ifpack2`)
- [x] `Tpetra` implementation of `Vector`
- [x] basic test infrastructure that instantiates tests for the enabled (S, L, G, N).
- [x] test of `Vector`
- [ ] add `MultiVector`, `Group`, ...
- [ ] ...

Discussed with @rppawlo and briefly with @cgcg at HPSFCon.

This is a draft. If you should have feedback, please do not hesitate.

Perhaps we could do a first review after the step "test of `Vector`"?

mayrmt commented:
@maartenarnst, this looks very interesting for us (and might help us to get around having to use `Thyra`). Thanks!

maartenarnst commented:
Hi @rppawlo. Following our discussion at HPSFCon, I've gone ahead and opened this PR for the Tpetra implementation of NOX. Would you have some time to take a look?

In its current state, the PR has a basic infrastructure for the ETI of the classes and tests for the Tpetra implementation, as well as the Tpetra implementation and corresponding tests of NOX Vector. And so Multivector, Group, ... are not currently in this PR.

Some questions at this stage:
- What should we do with the existing folders `src-tpetra` and `tests/tpetra`, which actually relate to the Thyra implementation? In this PR, I've moved the file that was in `src-tpetra` to `src-thyra`, and I've renamed the `tests/tpetra` as `tests/thyra`.
- The way I set up the ETI is that NOX now declares itself *overall* as a package that uses ETI to the Tribits system. This may be somewhat unsatisfactory because it's only the new `Tpetra` implementation that uses ETI, but the other parts of NOX do not. An alternative may be to try to have a sub package for the new `Tpetra` implementation. I looked a little bit into this possibility, but I'm not sure Tribits allows there to be a sub package, without the other parts also being reworked into sub packages. So it seems that making the new `Tpetra` implementation a sub package would require a very large refactoring of NOX more generally, and that may not be warranted. What do you think?
- At this stage, it may make sense to try to run this PR once through the testing pipelines to see what happens (probably error messages :)). Would you have a moment to set the labels?
- How do you think we should proceed? Should we go for detailed reviews of this PR in its current form, try to merge this part, and then bring the next pieces in new PRs? Or do you prefer that we first bring everything?

Thanks in advance!

cgcgcg reviewed (dismissed):
Approval for AT2 run

cgcgcg commented:
```
CMake Error at cmake/tribits/core/package_arch/TribitsAddExecutable.cmake:340 (add_executable):
  add_executable cannot create target "NOX_Tpetra_VectorOpsTests" because
  another target with the same name already exists.  The existing target is
  an executable created in source directory
  "/home/runner/_work/Trilinos/Trilinos/packages/nox/test/thyra".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:40 (tribits_add_executable)
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:202 (tribits_add_executable_wrapper)
  packages/nox/test/tpetra/CMakeLists.txt:6 (TRIBITS_ADD_EXECUTABLE_AND_TEST)
```

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; this inspection will remain valid until a new commit to source branch is performed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1585</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0d752e65265cfc33abc7917c854228c245c23d42</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1632</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0d752e65265cfc33abc7917c854228c245c23d42</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda</h3>
<ul>
  <li>Build Num: 1631</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8-gpu</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0d752e65265cfc33abc7917c854228c245c23d42</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1631</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0d752e65265cfc33abc7917c854228c245c23d42</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>
</details>



<br>[CDash Test Results for PR# 14084.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-14084-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

<br>[Wiki: How to Reproduce PR Testing Builds and Errors.](https://github.com/trilinos/Trilinos/wiki/Reproducing-PR-Testing-Errors)<br>

maartenarnst commented:
> ```
> CMake Error at cmake/tribits/core/package_arch/TribitsAddExecutable.cmake:340 (add_executable):
>   add_executable cannot create target "NOX_Tpetra_VectorOpsTests" because
>   another target with the same name already exists.  The existing target is
>   an executable created in source directory
>   "/home/runner/_work/Trilinos/Trilinos/packages/nox/test/thyra".  See
>   documentation for policy CMP0002 for more details.
> Call Stack (most recent call first):
>   cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:40 (tribits_add_executable)
>   cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:202 (tribits_add_executable_wrapper)
>   packages/nox/test/tpetra/CMakeLists.txt:6 (TRIBITS_ADD_EXECUTABLE_AND_TEST)
> ```

OK. Indeed. Sorry about that. I've just fixed it. We'll probably need to set the label again because the other tests probably failed for the same reason.

cgcgcg reviewed (dismissed):
AT2

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1586</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>4c595635505b60b255e53dfa5a4c88f1e53d067a</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1633</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>4c595635505b60b255e53dfa5a4c88f1e53d067a</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda</h3>
<ul>
  <li>Build Num: 1632</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8-gpu</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>4c595635505b60b255e53dfa5a4c88f1e53d067a</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1632</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>4c595635505b60b255e53dfa5a4c88f1e53d067a</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (uliegecsm/Trilinos)</dt>
  <ul>
    <li>Branch: nox-tpetra</li>
    <li>SHA: 4c595635505b60b255e53dfa5a4c88f1e53d067a</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maartenarnst
</details>

cgcgcg commented:
```
CMake Error at cmake/tribits/core/package_arch/TribitsAddExecutable.cmake:340 (add_executable):
  add_executable cannot create target "NOX_Thyra_1DFEM" because another
  target with the same name already exists.  The existing target is an
  executable created in source directory
  "/home/runner/_work/Trilinos/Trilinos/packages/nox/test/epetra/Thyra".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:40 (tribits_add_executable)
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:202 (tribits_add_executable_wrapper)
  packages/nox/test/thyra/CMakeLists.txt:36 (TRIBITS_ADD_EXECUTABLE_AND_TEST)
```

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Error: Jenkins Jobs - A user has pushed a change to the PR before testing completed. NEW EVENT 'committed', ID C_kwDOAsJyMdoAKGNmNGNhYzI1ODE3MTUzMjVjZTAzNzg1ZjVjODQ4Nzg3Y2Q4NzQ5MjQ... The Jenkins Jobs will be shutdown; Testing of this PR must occur again.

cgcgcg reviewed (approved):
AT2

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - User Requested Retest - Label AT: RETEST will be reset after testing.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; this inspection will remain valid until a new commit to source branch is performed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - AutoMerge IS ENABLED, but the Label AT: AUTOMERGE is not set.  Either set Label AT: AUTOMERGE or manually merge the PR...

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - AutoMerge IS ENABLED, but the Label AT: AUTOMERGE is not set.  Either set Label AT: AUTOMERGE or manually merge the PR...

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - AutoMerge IS ENABLED, but the Label AT: AUTOMERGE is not set.  Either set Label AT: AUTOMERGE or manually merge the PR...

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - AutoMerge IS ENABLED, but the Label AT: AUTOMERGE is not set.  Either set Label AT: AUTOMERGE or manually merge the PR...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

maartenarnst commented:
Hi @rppawlo, would you have a moment to take a look?

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

maartenarnst commented:
> Hi @rppawlo. Following our discussion at HPSFCon, I've gone ahead and opened this PR for the Tpetra implementation of NOX. Would you have some time to take a look?
> 
> In its current state, the PR has a basic infrastructure for the ETI of the classes and tests for the Tpetra implementation, as well as the Tpetra implementation and corresponding tests of NOX Vector. And so Multivector, Group, ... are not currently in this PR.
> 
> Some questions at this stage:
> 
> * What should we do with the existing folders `src-tpetra` and `tests/tpetra`, which actually relate to the Thyra implementation? In this PR, I've moved the file that was in `src-tpetra` to `src-thyra`, and I've renamed the `tests/tpetra` as `tests/thyra`.
> * The way I set up the ETI is that NOX now declares itself _overall_ as a package that uses ETI to the Tribits system. This may be somewhat unsatisfactory because it's only the new `Tpetra` implementation that uses ETI, but the other parts of NOX do not. An alternative may be to try to have a sub package for the new `Tpetra` implementation. I looked a little bit into this possibility, but I'm not sure Tribits allows there to be a sub package, without the other parts also being reworked into sub packages. So it seems that making the new `Tpetra` implementation a sub package would require a very large refactoring of NOX more generally, and that may not be warranted. What do you think?
> * At this stage, it may make sense to try to run this PR once through the testing pipelines to see what happens (probably error messages :)). Would you have a moment to set the labels?
> * How do you think we should proceed? Should we go for detailed reviews of this PR in its current form, try to merge this part, and then bring the next pieces in new PRs? Or do you prefer that we first bring everything?
> 
> Thanks in advance!

Hey @rppawlo, would you have a moment to take a look?

rppawlo commented:
@maartenarnst - apologies - was on vacation last week and am on travel again this week. I will look in detail on friday. 

1. I'm not sure about moving the code. We could potentially reuse some of the test code so that we don't have to repeat it.
2. I'm fine with having ETI. At some point we would like to explore complex nonlinear solves so this would be needed for that as well.
3. committing in separate stages is my preference, but either way is fine with me. Will launch the PR testing shortly.

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

maartenarnst commented:
Hey @rppawlo, would you have time to take a look this week?

rppawlo reviewed (changes requested):
@maartenarnst - thanks for adding this code! In addition to the minor comments above, could you make one other change? For the code that was moved out of src-tpetra and tests/tpetra, could these be moved into src-thyra-tpetra and tests/thyra-tpetra respectively? Thyra is an abstract layer and could have multiple concrete implementations underneath, so I would like to keep the tpetra specific code for thyra separate from the generic thyra code.

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES!

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However PR is now STALE, and must be retested.  Set the AT: RETEST Label to force retest....

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

rppawlo commented on packages/nox/src-tpetra/NOX_Tpetra_Vector_def.hpp:
```diff
@@ -0,0 +1,353 @@
+Abstract::Vector& Vector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+random(bool useSeed, int seed)
+{
+    if (!tpetraVec->isDistributed())
+    {
+        if (tpetraVec->getMap()->getComm()->getRank() == 0) {
+            tpetraVec->randomize(-Teuchos::ScalarTraits<Scalar>::one(), Teuchos::ScalarTraits<Scalar>::one());
+        }
+        else {
+            tpetraVec->putScalar(Teuchos::ScalarTraits<Scalar>::zero());
+        }
+        tpetraVec->reduce();
+    }
+    else {
```
I don't think we need it for NOX, but there might a use case in LOCA. Let's leave it in for now.

rppawlo commented on packages/nox/src-tpetra/NOX_Tpetra_Vector_def.hpp:
```diff
@@ -0,0 +1,353 @@
+} // namespace NOX::Tpetra
+
+//! Explicit template instantiation macro for @ref NOX::Tpetra::Vector.
+#define NOX_TPETRA_VECTOR_INSTANT(S, L, G, N) template class NOX::Tpetra::Vector<S, L, G, N>;
```
Let's keep it the way it is. In panzer, we decided to only support certain scalar types and hard code to always use ETI. It eliminated a lot of the complexity. But since we may want complex scalar types in the future, I think we need the full machinery for nox.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1716</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>bea5cfe13c7d4dfee34aa4d267e348238bec9ef8</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1763</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>bea5cfe13c7d4dfee34aa4d267e348238bec9ef8</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1762</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>bea5cfe13c7d4dfee34aa4d267e348238bec9ef8</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_CompSim</h3>
<ul>
  <li>Build Num: 64</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_aue-gcc-openmpi_release_static_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_compsim</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>bea5cfe13c7d4dfee34aa4d267e348238bec9ef8</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a0c3c90a9012ecc0cff2d87f593c48fce25ffc78</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (uliegecsm/Trilinos)</dt>
  <ul>
    <li>Branch: nox-tpetra</li>
    <li>SHA: bea5cfe13c7d4dfee34aa4d267e348238bec9ef8</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maartenarnst
</details>

maartenarnst commented:
Hi @rppawlo. Thanks for approving. It looks like the "DCO" test failed because not all commits were signed off. So I've just made sure to sign them all. Would you have a moment to set the labels again?

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1718</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1765</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1764</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_CompSim</h3>
<ul>
  <li>Build Num: 66</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_aue-gcc-openmpi_release_static_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_compsim</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (uliegecsm/Trilinos)</dt>
  <ul>
    <li>Branch: nox-tpetra</li>
    <li>SHA: 01d4bf54fc5d085e041a359e5c5b652aec74b445</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maartenarnst
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1718</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1765</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1764</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_CompSim</h3>
<ul>
  <li>Build Num: 66</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_aue-gcc-openmpi_release_static_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_compsim</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>01d4bf54fc5d085e041a359e5c5b652aec74b445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>b2faff61341f586d2306d2a27f2923b081ff604c</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 14084.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-14084-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1733</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1780</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1779</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_CompSim</h3>
<ul>
  <li>Build Num: 81</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_aue-gcc-openmpi_release_static_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_compsim</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (uliegecsm/Trilinos)</dt>
  <ul>
    <li>Branch: nox-tpetra</li>
    <li>SHA: c403b1b78769d92aa5b3db7be734732962b2a445</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maartenarnst
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: PR_gcc-openmpi-openmp</h3>
<ul>
  <li>Build Num: 1733</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-gnu-8.5.0-openmpi-4.1.6-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_clang</h3>
<ul>
  <li>Build Num: 1780</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-clang-11.0.1-openmpi-4.0.5-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_cuda-uvm</h3>
<ul>
  <li>Build Num: 1779</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_sems-cuda-11.4.2-gnu-10.1.0-openmpi-4.1.6_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: PR_CompSim</h3>
<ul>
  <li>Build Num: 81</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>BLOCKING_BUILD</td>
    <td>false</td>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel8_aue-gcc-openmpi_release_static_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_compsim</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: NOX;AT: AUTOMERGE</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>14084</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>rhel8</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/uliegecsm/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c403b1b78769d92aa5b3db7be734732962b2a445</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fc801029a273d4c8c751e98f9a3993ba11781e96</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 14084.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-14084-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ rppawlo ]!

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Cannot Merge Pull Request# 14084: Github Mergeability = blocked; Git Mergeable = True

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Merge on Pull Request# 14084: IS A SUCCESS - Pull Request successfully merged
